src/convertToObject.js

```diff
@@ -6,7 +6,19 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  if (!sourceString.trim()) {
+    return {};
+  }
+
+  return sourceString
+    .split(';')
+    .map((keyValue) => keyValue.split(':').map((str) => str.trim()))
+    .filter(([key, value]) => key && value)
```

Comment:
The filter method is correctly used to ensure that both key and value exist. This prevents adding undefined or empty keys/values to the resulting object.